doc/specifications/proposed/rest_resource_embedding.md

```diff
@@ -0,0 +1,173 @@
+</Content>
+```
+
+By adding an `X-eZ-Embed-Value` header to the request, we can get the
+main location object embedded into the response:
+
+```
+curl -X GET http://localhost:8000/api/ezp/v2/content/objects/1 -H 'Cookie: XDEBUG_SESSION=PHPSTORM' -H 'x-ez-embed-value: Content.MainLocation'                 
+```
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<Content media-type="application/vnd.ez.api.ContentInfo+xml" href="/api/ezp/v2/content/objects/1" remoteId="9459d3c29e15006e45197295722c7ade" id="1">
+ <ContentType media-type="application/vnd.ez.api.ContentType+xml" href="/api/ezp/v2/content/types/1"/>
+ <Name>eZ Platform</Name>
+ <Versions media-type="application/vnd.ez.api.VersionList+xml" href="/api/ezp/v2/content/objects/1/versions"/>
+ <CurrentVersion media-type="application/vnd.ez.api.Version+xml" href="/api/ezp/v2/content/objects/1/currentversion"/>
+ <Section media-type="application/vnd.ez.api.Section+xml" href="/api/ezp/v2/content/sections/1"/>
+ <MainLocation media-type="application/vnd.ez.api.Location+xml" href="/api/ezp/v2/content/locations/1/2">
+  <id>2</id>
+  <priority>0</priority>
+  <hidden>false</hidden>
+  <invisible>false</invisible>
+  <ParentLocation media-type="application/vnd.ez.api.Location+xml" href="/api/ezp/v2/content/locations/1"/>
+  <pathString>/1/2/</pathString>
+  <depth>1</depth>
+  <childCount>8</childCount>
+  <remoteId>f3e90596361e31d496d4026eb624c983</remoteId>
+  <Children media-type="application/vnd.ez.api.LocationList+xml" href="/api/ezp/v2/content/locations/1/2/children"/>
+  <Content media-type="application/vnd.ez.api.Content+xml" href="/api/ezp/v2/content/objects/1"/>
+  <sortField>PRIORITY</sortField>
+  <sortOrder>ASC</sortOrder>
+  <UrlAliases media-type="application/vnd.ez.api.UrlAliasRefList+xml" href="/api/ezp/v2/content/locations/1/2/urlaliases"/>
+  <ContentInfo media-type="application/vnd.ez.api.ContentInfo+xml" href="/api/ezp/v2/content/objects/1"/>
+ </MainLocation>
+ <Locations media-type="application/vnd.ez.api.LocationList+xml" href="/api/ezp/v2/content/objects/1/locations"/>
+ <Owner media-type="application/vnd.ez.api.User+xml" href="/api/ezp/v2/user/users/14"/>
+ <lastModificationDate>2015-11-30T13:10:46+00:00</lastModificationDate>
+ <publishedDate>2015-11-30T13:10:46+00:00</publishedDate>
+ <mainLanguageCode>eng-GB</mainLanguageCode>
+ <currentVersionNo>9</currentVersionNo>
+ <alwaysAvailable>true</alwaysAvailable>
+ <ObjectStates media-type="application/vnd.ez.api.ContentObjectStates+xml" href="/api/ezp/v2/content/objects/1/objectstates"/>
+</Content>
+```
+
+### The `X-eZ-Embed-Value` request header
+
+Which resources must be embedded is specified using this request header.
+It accepts several resources separated by commas:
+
+`X-eZ-Embed-Value: Content.MainLocation,Content.Owner.Groups`
+
+A resource is referenced by its "path" from the root of the response.
+Resources from an embedded resource can also be embedded:
```

Comment:
Do we need some form of recursion protection? I mean, to avoid intentional or accidental DOS-ing with e.g. `Content.MainLocation.Content.MainLocation.Content.MainLocation` etc? At least we can't have an "expand all" feature.